lock-api/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -40,10 +40,18 @@ default boolean isInitialized() {
 
     TimestampRange getFreshTimestamps(@Safe @QueryParam("number") int numTimestampsRequested);
 
+    /**
+     * @deprecated use {@link #lockImmutableTimestamp()} instead.
+     */
+    @Deprecated
+    default LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request) {
+        return lockImmutableTimestamp();
+    }
```

Comment:
I'm not sure if it's better if we keep this method around or just force a dev break on `lockImmutableTimestamp` users, since the semantics of this are different from the previous method.

Reply:
I am more inclined towards forcing a dev break - given that this should have a very narrow use.

Reply:
agreed